src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -140,8 +142,13 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             }
             else
             {
+                if (!m_acceptSourceAgreements && SourceAgreements().Size() != 0)
+                {
+                    return GetConnectSourceAgreementsNotAcceptedErrorResult();
+                }
+
                 source = m_sourceReference;
-                source.SetCaller(GetCallerName());
+                source.SetCaller(callerName);

```

Comment:
Don't you need to call `SetBackgroundUpdateInterval` here?  I'm not seeing it being set anywhere else...